### PR TITLE
feat: [PLG-2131]: marketplace provision licenses

### PR DESCRIPTION
**Pull request by mtran7**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47031)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/resources/UserResourceNG.java [hunk 1]
```diff
@@ -189,6 +208,54 @@ public RestResponse<UserInfo> createCommunityUserAndCompleteSignup(SignupInviteD
...
+          EXISTING_INVITE_ALREADY_COMPLETED, ErrorCode.USER_INVITE_OPERATION_FAILED, WingsException.USER);
+    }
+
+    // String email = dto.getEmail().toLowerCase();
+
+    userService.verifyRegisteredOrAllowed(email);
+
+    // dto.setEmail(dto.getEmail().toLowerCase());
+
+    User user = convertMarketplaceRequestToUser(email, password, account);
```
Comment (jamie-harness) [inline 400-rest/src/main/java/software/wings/resources/UserResourceNG.java:230]: this user doesn't seem to be used. Is there some thing missing?

**Comment by jamie-harness** [comment]
It seems it's still WIP, can you convert it to draft or label it as WIP?

File: 400-rest/src/main/java/software/wings/resources/UserResourceNG.java (hunk 1 above)
Comment (mtran7) [reply 400-rest/src/main/java/software/wings/resources/UserResourceNG.java:230]: User is being used in lines 250 and 251 

```
    String accountId = userService.setupAccountBasedOnProduct(user, userInvite, marketPlace);
    User createdUser = userService.createNewUserAndSignIn(user, accountId, NG);
```

**Comment by mtran7** [comment]
trigger utAll

**Comment by mtran7** [comment]
trigger utAll

**Pull request closed without merging**